### DISCUSSION TITLE
Possibility Then-flow Card for "prohibit" and "allow" for Domestic Hot Water

### DISCUSSION THREAD

**Discussion by ekruijf** [OlivierZal/com.melcloud#194] [Ideas]
I enjoy using this app. The possibilities would be increased if a Then-flow Card became available that allows you to switch between "prohibit" and "allow" for Domestic Hot Water and also for Heating and Cooling. This way you can determine in Homey when these operating modes should become active instead of having to use the weekly clock.

One of the things I would like to do is to have certain programs only become active at the hottest point of the day, for example.

Does anyone know if this is possible? And if so, can I help with anything to make this happen?

**Comment by OlivierZal** [comment]
Hi @ekruijf 

I think it's possible to know when a device runs under this status.

However it's not possible to set this status, since this action is not available from the MELCloud API but only from the physical device (or maybe remote control). 

In the purpose to make this status visible I would need to have an access to your device since myself have an air-to-air heat pump.
Can you please provide olivier.zalmanski@me.com with a guest access from the official MELCloud (mobile or web) app?

Kind regards

**Comment by ekruijf** [comment]
Hi @OlivierZal ,

Thanks for your quick response!

I can already see the current operating mode. But thanks for your offer :)

What I really care about is setting the status so that I can determine when the Heat Pump will perform it. I don't have a remote, just a physical unit on the wall in the living room. This allows me to indeed set the statuses "forbid" and "Allowed" for each mode such as Domestic hot water and Heating.

Do you think there is another way to switch on - for example sanitary hot water - at times determined in Homey? I first wanted to do it with forced hot water, but then electric heating elements are also used, while the aim is to save energy.

Kind regards

**Comment by OlivierZal** [comment]
Hi @ekruijf,

I'm sure it's not possible to set forbid/allow from Homey because I make available all the possible MELCloud API actions.

In term of display indeed Homey displays the "current status" (idle, heating, domestic hot water, etc.) but there is another information "prohibit" (true/false) which I didn't use for the moment, because it complexifies the dashboard but I can add it as optional from settings.

Maybe an air-to-water expert would know how to perform what you want, but since I don't have one... maybe someone could help you on the dedicated forum? https://community.homey.app/t/app-pro-melcloud/73094/31

**Comment by OlivierZal** [comment]
Hi @ekruijf

After investigation I still confirm that the prohibit / unprohibit action is not possible.

Concerning a possible prohibit status display I will give up on this for at least 2 reasons:
- there are 5 available values: hot water, heating/cooling zone1/zone2, what is complicated to handle (display + UX)
- this is not a status "per se" but rather a manual blocker, and fact that it is not possible to control it except from the physical device doesn't really fit the Homey / domotic automation philosophy
